### PR TITLE
Adds builds_external ddoc and tests

### PR DESCRIPTION
**Pull request by dianabarsan**
#14

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "ecmaVersion": 2018
   },
   "rules": {
-    "no-console": 0
+    "no-console": 0,
+    "semi": ["error", "always"]
```
Comment (dianabarsan) [inline .eslintrc:12]: Added https://github.com/medic/medic-builds-repo/issues/19 to change base eslint rules.

**Comment by dianabarsan** [comment]
Hello @craig-landry . 
I'm trying to unblock external contributions from being unable to pass CI (specifically blocking this work in progress: https://github.com/medic/cht-core/pull/7422). If you don't have enough time to have a look, I can wait until Gareth gets back. 

Thanks!

File: scripts/pushToServer.js [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
+
+  console.log('..pushing generated ddoc');
+  await DB.put(compiledDdoc);
+  console.log('..done');
+};
+
+(async function main() {
+  console.log('Compiling ddocs');
+  const ddoc = await compile(path.join(__dirname, '..', 'ddocs/builds'));
+  const buildsExternalDdoc = await compile(path.join(__dirname, '..', 'ddocs/builds_external'));
```
Comment (craig-landry) [inline scripts/pushToServer.js]: Just separating out the path building and giving things names for easy skimming.

```suggestion
  const buildsDir = path.join(__dirname, '..', 'ddocs/builds');
  const externalBuildsDir = path.join(__dirname, '..', 'ddocs/builds_external');
  
  const ddoc = await compile(buildsDir);
  const buildsExternalDdoc = await compile(externalBuildsDir);
```

File: scripts/pushToServer.js [hunk 3]
```diff
@@ -1,48 +1,48 @@
...
+
+  console.log('..getting existing ddoc (for _rev)');
+  try {
+    const existing = await DB.get(compiledDdoc._id);
+    if (existing) {
+      compiledDdoc._rev = existing._rev;
+    }
+  } catch (err) {
+    delete compiledDdoc._rev;
+    console.log('No existing doc');
```
Comment (craig-landry) [inline scripts/pushToServer.js:24]: What's the case where the `if (existing)` above is false that doesn't end up here instead?

File: test/int/builds-external-ddoc.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    author: 'int. test',
+    node_modules: []
+  };
+
+  return doc;
+};
+
+describe('"Builds External" Design document', () => {
+  before(() => resetDb());
+  afterEach(() => clearDb());
```
Comment (craig-landry) [inline test/int/builds-external-ddoc.js]:
```suggestion
  before(resetDb);
  afterEach(clearDb);
```

File: test/int/builds-external-ddoc.js [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+      await buildsDb.remove({ _id: buildId, _rev: rev });
+    });
+
+    it('should now allow creation of documents with invalid ids', async () => {
+      try {
+        await buildsDb.put(withBuildInfo({ _id: 'not-a-valid-version-identifier' }));
+        expect.fail('Should have thrown');
+      } catch (err) {
+        expect(err.message).to.equal('Document _id format invalid');
+      }
```
Comment (craig-landry) [inline test/int/builds-external-ddoc.js]: I'd like to use chai-as-promised and do this with `to.be.rejectedWith` to remove the try/catch business here. Given that's not in our codebase yet, I'll skip on it for now and leave this alone, but I really would like to get this (and all the other cases in our tests in general) to be logical chai checks and not have the try/catch here.

File: test/int/builds-external-ddoc.js [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+      };
+      try {
+        await buildsDb.put(doc);
+        expect.fail('Should have thrown');
+      } catch (err) {
+        expect(err.message).to.match(/Incompatible schema_version/);
+      }
+    });
+  });
+
```
Comment (craig-landry) [inline test/int/builds-external-ddoc.js]:
```suggestion
```

**Review by craig-landry — CHANGES_REQUESTED** [review]

File: scripts/pushToServer.js (hunk 3 above)
Comment (dianabarsan) [reply scripts/pushToServer.js:24]: We just upload with no required changes.

Comment (dianabarsan) [reply scripts/pushToServer.js:24]: Ah I get your point! :)

File: test/int/builds-external-ddoc.js (hunk 5 above)
Comment (dianabarsan) [reply test/int/builds-external-ddoc.js]: Turns out chai-as-promised was already in the package :)

Comment (craig-landry) [reply test/int/builds-external-ddoc.js]: Oh, interesting. I grepped for it, but must have done something wrong. Anyway, do my chai-as-promised sales pitch in the future :)

**Review by craig-landry — APPROVED** [review]

**Pull request merged**